end_to_end/2-lineage-train-assess-bias-tune-registry-e2e.ipynb

```diff
@@ -0,0 +1,1083 @@
+{
```

Comment:
<p>Got some error when execute <code>clarify_processor.run_bias</code></p><pre class="ql-syntax" spellcheck="false">Traceback (most recent call last):
&nbsp; File "/usr/local/lib/python3.7/site-packages/smclarify/bias/report.py", line 294, in _categorical_metric_call_wrapper
&nbsp; &nbsp; group_variable=group_variable,
&nbsp; File "/usr/local/lib/python3.7/site-packages/smclarify/bias/metrics/__init__.py", line 26, in call_metric
&nbsp; &nbsp; return metric(**{key: kwargs[key] for key in inspect.signature(metric).parameters.keys()})
&nbsp; File "/usr/local/lib/python3.7/site-packages/smclarify/bias/metrics/pretraining.py", line 195, in CDDL
&nbsp; &nbsp; return common.CDD(feature, sensitive_facet_index, positive_label_index, group_variable)
&nbsp; File "/usr/local/lib/python3.7/site-packages/smclarify/bias/metrics/common.py", line 80, in CDD
&nbsp; &nbsp; raise ValueError("Group variable is empty or not provided")
</pre><p> </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/1974/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Reply:
<p>Mine is running fine...</p><pre class="ql-syntax" spellcheck="false">

Job Name:  Clarify-Bias-2021-02-07-07-47-08-632
Inputs:  [{'InputName': 'dataset', 'AppManaged': False, 'S3Input': {'S3Uri': 's3://sagemaker-us-east-2-738335684114/fraud-detect-demo/data/train/train.csv', 'LocalPath': '/opt/ml/processing/input/data', 'S3DataType': 'S3Prefix', 'S3InputMode': 'File', 'S3DataDistributionType': 'FullyReplicated', 'S3CompressionType': 'None'}}, {'InputName': 'analysis_config', 'AppManaged': False, 'S3Input': {'S3Uri': 's3://sagemaker-us-east-2-738335684114/Clarify-Bias-2021-02-07-07-47-08-632/input/analysis_config/analysis_config.json', 'LocalPath': '/opt/ml/processing/input/config', 'S3DataType': 'S3Prefix', 'S3InputMode': 'File', 'S3DataDistributionType': 'FullyReplicated', 'S3CompressionType': 'None'}}]
Outputs:  [{'OutputName': 'analysis_result', 'AppManaged': False, 'S3Output': {'S3Uri': 's3://sagemaker-us-east-2-738335684114/fraud-detect-demo/clarify-output/bias_1', 'LocalPath': '/opt/ml/processing/output', 'S3UploadMode': 'EndOfJob'}}]
.................................INFO:sagemaker-clarify-processing:Starting SageMaker Clarify Processing job
INFO:analyzer.data_loading.data_loader_util:Analysis config path: /opt/ml/processing/input/config
INFO:analyzer.data_loading.data_loader_util:Analysis result path: /opt/ml/processing/output
INFO:analyzer.data_loading.data_loader_util:This host is algo-1.
INFO:analyzer.data_loading.data_loader_util:This host is the leader.
INFO:analyzer.data_loading.data_loader_util:Number of hosts in the cluster is 1.
INFO:sagemaker-clarify-processing:Running Python / Pandas based analyzer.
INFO:analyzer.data_loading.data_loader_util:Dataset uri: /opt/ml/processing/input/data
INFO:analyzer.data_loading.data_loader_util:Dataset type: text/csv
INFO:analyzer.predictor:Spinning up shadow endpoint
INFO:sagemaker:Creating endpoint-config with name sagemaker-clarify-endpoint-config-1612684349-6766
INFO:analyzer.predictor:Creating endpoint sagemaker-clarify-endpoint-1612684349-e078
INFO:sagemaker-clarify-processing:======================================
INFO:sagemaker-clarify-processing:Calculating post-training bias metrics
INFO:sagemaker-clarify-processing:======================================
INFO:sagemaker-clarify-processing:Getting predictions from the endpoint
INFO:analyzer.predictor:Checking endpoint status
</pre><p> </p>